test/google_search_data_viewer/accounts/schemas/user_test.exs

```diff
@@ -0,0 +1,53 @@
+defmodule GoogleSearchDataViewer.Accounts.Schemas.UserTest do
+  use GoogleSearchDataViewer.DataCase, async: true
+
+  alias GoogleSearchDataViewer.Accounts.Schemas.User
+
+  describe "changeset/2" do
+    test "given an empty changeset with empty fields, fails to validate" do
+      user_changeset = User.changeset(%User{}, %{})
+
+      refute user_changeset.valid?
+    end
+
+    test "given a changeset with email and password fields, validates" do
+      user_changeset =
+        User.changeset(%User{}, %{email: "testemail@email.com", password: "aValidPassword"})
+
+      assert user_changeset.valid?
+    end
+  end
+
+  describe "register_changeset/2" do
+    test "given an invalid email format, shows validation error for email field" do
+      user_changeset =
+        User.register_changeset(%User{}, %{email: "testemail", password: "aValidPassword"})
+
+      refute user_changeset.valid?
+
+      assert %{email: ["must have the @ sign and no spaces"]} = errors_on(user_changeset)
+    end
+
+    test "given a pre-registered email, shows validation error for unique email constraint" do
+      insert(:user, email: "duplicate@email.com", password: "aValidPassword")
+
+      user_changeset =
+        User.register_changeset(%User{}, %{email: "duplicate@email.com", password: "aValidPassword"})
+
+      assert {:error, user_changeset} = Repo.insert(user_changeset)
+
+      refute user_changeset.valid?
+
+      assert %{email: ["has already been taken"]} = errors_on(user_changeset)
```

Comment:
![Screen Shot 2022-05-10 at 09 40 11](https://user-images.githubusercontent.com/948856/167531934-2671ddf7-12c2-444c-bc66-5fdaa4a66f0b.png)

https://nimblehq.co/compass/development/code-conventions/elixir/ex-unit/

Reply:
Using double equals to test for equality is still preferred over pattern matching when I actually only want to test for partial matches, ie the email key anyway?